app/models/course.rb

```diff
@@ -190,6 +190,10 @@ def new_draft_attributes
     where(qualification: qualifications)
   end
 
+  scope :with_provider_name, ->(provider_name) do
+    where(provider_id: Provider.with_full_name(provider_name).select(:id))
```

Comment:
```suggestion
    joins(:providers).merge(Provider.with_full_name(provider_name))
```

Reply:
Can I ask what this change is for?

Reply:
```
[5] pry(main)> Course.where(provider_id: Provider.with_full_name("dave").select(:id)).to_sql
=> "SELECT \"course\".* FROM \"course\" WHERE \"course\".\"provider_id\" IN (SELECT \"provider\".\"id\" FROM \"provider\" WHERE \"provider\".\"provider_name\" = 'dave')"
[6] pry(main)> Course.joins(:provider).merge(Provider.with_full_name("dave")).to_sql
=> "SELECT \"course\".* FROM \"course\" INNER JOIN \"provider\" ON \"provider\".\"id\" = \"course\".\"provider_id\" WHERE \"provider\".\"provider_name\" = 'dave'"
```

I think @faissalMT 's suggestion comes out with the cleaner (and likely faster) SQL here. I can't remember why I didn't use `merge` on some of the earlier scopes. I think it was something to do with it not chaining properly when we join them together for the course search. I don't think that's an issue with `merge` though, it was probably to do with what was being merged.

Reply:
@SamJamCul It doesn't make a huge difference here since you're (almost always) just picking out one item, but as Graham said doing things this way tends to be faster and more straightforward.

Reply:
definitely better sql there

Reply:
Some stackoverflow discussion - https://stackoverflow.com/questions/50490009/rails-activerecord-where-vs-merge

Reply:
Changed!